File: src/main/java/io/teammion/morefood/Registry.java
package io.teammion.morefood;

import net.minecraft.block.Block;
import net.minecraft.client.Minecraft;
import net.minecraft.client.renderer.block.model.ModelResourceLocation;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraftforge.fml.common.registry.GameRegistry;

/**
 * Created by Stefan Wimmer <stefanwimmer128@gmail.com> on 05.06.16.
 */
public class Registry extends GameRegistry
{
    public static void render(Item item)
    {
        render(item, 0, item.getUnlocalizedName().substring(5));
    }
    
    public static void render(Item item, int meta, String modelResourceLocation)
    {
        Minecraft.getMinecraft().getRenderItem().getItemModelMesher().register(
                item, meta, new ModelResourceLocation(
                        "morefood:" + modelResourceLocation, "inventory"
                )
        );
    }
    
    public static void addSmelting(Item item, ItemStack stack)
    {
        addSmelting(item, stack, .8f);
    }
    
    public static void addSmelting(ItemStack stack0, ItemStack stack1)
    {
        addSmelting(stack0, stack1, .8f);
    }
    
    public static void addSmelting(Block item, ItemStack stack)
    {
        addSmelting(item, stack, .8f);
    }
}

File: src/main/scala/io/teammion/morefood/Registry.scala
package io.teammion.morefood

import net.minecraft.client.Minecraft
import net.minecraft.client.renderer.block.model.ModelResourceLocation
import net.minecraft.item.{Item, ItemStack}
import net.minecraftforge.fml.common.registry.GameRegistry

/**
  * Created by Stefan Wimmer <stefanwimmer128@gmail.com> on 24.07.16.
  */
object Registry
{
    def register(item : Item) : Unit =
        GameRegistry.register(item)
    
    def render(item : Item, meta : Int = 0, name : String = null) : Unit =
        Minecraft.getMinecraft.getRenderItem.getItemModelMesher.register(
            item, meta, new ModelResourceLocation(
                "tm-morefood:" + Option(name).getOrElse(item.getUnlocalizedName.substring(5)), "inventory"
            )
        )
    
    def addSmelting(in : Item, out : ItemStack) =
        GameRegistry.addSmelting(in, out, .2f)
    
    def addSmelting(in : ItemStack, out : ItemStack) =
        GameRegistry.addSmelting(in, out, .2f)
    
    def addShapelessRecipe(out : ItemStack, params : AnyRef*) : Unit =
        GameRegistry.addShapelessRecipe(out, params)
    
    def addShapedRecipe(out : ItemStack, params : AnyRef*) =
        GameRegistry.addShapedRecipe(out, params)
}
